### PR TITLE
Feature/country as entity

### PR DESCRIPTION
**Pull request by diegovalerad**
Country field from cars is now an entity, with its own REST layer, service, etc. Unit tests & integration tests updated and passed.

### CODE REVIEW

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplBrand.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	 * @param brand Name of the brand
+	 * @param company Company of the brand
+	 * @return Boolean
+	 */
+	public boolean brandNameAndCompanyExists(String brand, String company) {
+		String query = "SELECT brand "
+						+ "FROM Brand brand "
+						+ "WHERE brand.company='" + company + "' "
+								+ "AND brand.brand='" + brand + "'";
+		TypedQuery<Brand> createQuery = em.createQuery(query, Brand.class);
```
Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplBrand.java]: this should be better done using criteriaquery

Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplBrand.java]: I would do this better using criteriaQuery

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCar.java [hunk 2]
```diff
@@ -20,9 +20,11 @@
...
 	 */
 	public List<Car> getAllCarsFromCountry(String country){
-		String countryLowerCase = country.toLowerCase();
-		
-		String query = "SELECT car FROM Car car WHERE car.country='" + countryLowerCase + "'";
+		String query = "SELECT car "
+					+ "FROM Car car "
+					+ "WHERE car.country = (SELECT country "
+											+ "FROM Country country "
+											+ "WHERE country.countryName = '" + country  + "')";
```
Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCar.java]: I would do this better using criteriaQuery

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCountry.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	 * Checks if a country exists
+	 * @param countryName Name of the country to check
+	 * @return Boolean
+	 */
+	public boolean countryAlreadyExists(String countryName) {
+		String query = "SELECT country "
+					+ "FROM Country country "
+					+ "WHERE country.countryName='" + countryName + "'";
+		TypedQuery<Country> createQuery = em.createQuery(query, Country.class);
+
```
Comment (javieraviles) [inline src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCountry.java]: I would do this better using criteriaQuery

**Review by javieraviles — CHANGES_REQUESTED** [review]
I like this!!!

some details:
1) criteriaQuery for dynamic queries works better surely
2) check possibility to use mockito verify in unit tests
3) check logging level; not only info but warning/error are levels you should definitely aim for

good job!

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplBrand.java (hunk 1 above)
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplBrand.java]: done

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCar.java (hunk 2 above)
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCar.java]: done

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCountry.java (hunk 3 above)
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImplCountry.java]: done

**Pull request merged**